SSA/Projects/InstCombine/ForMathlib.lean

```diff
@@ -557,11 +557,51 @@ def beq {w : Nat} (x y : Bitvec w) : Bool := x = y
 
 def ofBool : Bool → Bitvec 1 := fun b => b ::ᵥ Vector.nil
 
-@[simp]
-def ofInt' {w} (i : Int) := Bitvec.ofInt w i
+def toBool : Bitvec 1 → Bool 
+  | b ::ᵥ Vector.nil => b
+
+theorem ofBool_toBool : ∀ b : Bool, toBool (ofBool b) = b := by simp only [ofBool, toBool]
+theorem toBool_ofBool : ∀ b : Bitvec 1, ofBool (toBool b) = b := by simp only [ofBool, toBool]
 
 instance : Coe Bool (Bitvec 1) := ⟨ofBool⟩
 
+
+instance : Coe (Bitvec 1) Bool := ⟨toBool⟩
+
+instance decPropToBitvec1 (p : Prop) [Decidable p] : CoeDep Prop p (Bitvec 1) where
+  coe := ofBool $ decide p
+
+
 infixl:75 ">>>ₛ" => fun x y => Bitvec.sshr x (Bitvec.toNat y)
 
+def ult (x y : Bitvec w) : Prop := x.toNat < y.toNat
+def ule (x y : Bitvec w) : Prop := x.toNat ≤ y.toNat
+def ugt (x y : Bitvec w) : Prop := x.toNat > y.toNat
+def uge (x y : Bitvec w) : Prop := x.toNat ≥ y.toNat
+
+instance {w : Nat} (x y : Bitvec w) : Decidable (ult x y) := by apply Nat.decLt
```

Comment:
I am curious where the suggested change works: 
```suggestion
instance {w : Nat} (x y : Bitvec w) : Decidable (ult x y) := by decide
```

Reply:
that's kind of janky, why does that even work? 😅 anyway, don't you think it's better to keep the more explicit term, if it's about the same size to write?

Reply:
@goens It works because `decide`, IIUC perform typeclass search for a `Decidable` instance. I'd guess even `infer_instance` might do the trick here, if it's able to reduce `ult x y` into `x.toNat > y.toNat`